src/api/migraphx.py

```diff
@@ -450,4 +451,9 @@ def experimental_custom_op(h):
     h.virtual('compute_shape',
               api.params(inputs='std::vector<migraphx::shape>'),
               returns='migraphx::shape')
+    # TODO : If output aliasing then, runs_on_offload_flag must be true
```

Comment:
This is not true. You could have output aliasing and run on the host. A CPU custom op would run on the host and take CPU output buffers.

Reply:
If compile target is GPU  and
If the output is aliasing and runs_on_offload_target is false, then MIGraphX would inject buffer copies to and from GPU memory. in that case, it is not clear if  custom_op would alias CPU buffer or GPU buffer.

Reply:
> in that case, it is not clear if custom_op would alias CPU buffer or GPU buffer.

It would alias the CPU buffer since that is the input. Right now, we naively copy back and forth for CPU operators but we could in the future only copy once when there is a series of CPU operators, so the aliased output buffer would become input to another CPU operator.

Reply:
Agree that buffer copy logic needs to be smarter. We would need that for FPGA partitioning also. 


> It would alias the CPU buffer since that is the input. 
Okay, but I guess wouldn't make sense if the consumer op of it is running on GPU. 

Do you want me to remove TODO?

Reply:
Removed TODO